roborock/web_api.py

```diff
@@ -332,6 +332,29 @@ async def get_home_data_v2(self, user_data: UserData) -> HomeData:
         else:
             raise RoborockException("home_response result was an unexpected type")
 
+    async def get_home_data_v3(self, user_data: UserData) -> HomeData:
+        """This is the same as get_home_data, but uses a different endpoint and includes non-robotic vacuums."""
+        rriot = user_data.rriot
+        if rriot is None:
+            raise RoborockException("rriot is none")
+        home_id = await self._get_home_id(user_data)
+        if rriot.r.a is None:
+            raise RoborockException("Missing field 'a' in rriot reference")
+        home_request = PreparedRequest(
+            rriot.r.a,
+            {
+                "Authorization": self._get_hawk_authentication(rriot, "/v3/user/homes/" + str(home_id)),
+            },
+        )
+        home_response = await home_request.request("get", "/v3/user/homes/" + str(home_id))
+        if not home_response.get("success"):
+            raise RoborockException(home_response)
+        home_data = home_response.get("result")
+        if isinstance(home_data, dict):
+            return HomeData.from_dict(home_data)
+        else:
+            raise RoborockException("home_response result was an unexpected type")
```

Comment:
```suggestion
        home_data = home_response.get("result")
        if isinstance(home_data, dict):
            return HomeData.from_dict(home_data)
        raise RoborockException(f"home_response result was an unexpected type: {home_data}")
```

Reply:
Thinking its helpful to have details about the failture ("I wonder what type it was then?")